skyportal/handlers/api/source.py

```diff
@@ -740,6 +731,14 @@ def get_sources(
                 "Invalid commentsFilterAuthor value -- must provide at least one string value"
             )
     if localization_dateobs is not None:
+
+        # This grabs just the IDs so the more expensive localization in-out
+        # check is done on only this subset
+        obj_ids = session.scalars(obj_query).all()
+        obj_query = Obj.select(user, options=obj_query_options).where(
+            Obj.id.in_(obj_ids)
+        )
+
         if localization_name is not None:
             localization = (
                 Localization.query_records_accessible_by(user)
```

Comment:
You forgot one `query_records_accessible_by`

Reply:
oops.